extensions/sql-database-projects/src/models/project.ts

```diff
@@ -609,102 +621,23 @@ export class Project implements ISqlProject {
-				await this.updateSystemDatabaseReferencesInProjFile();
-			}
-		} else if (this.containsSSDTOnlySystemDatabaseReferences()) {
-			const result = await window.showWarningMessage(constants.updateProjectDatabaseReferencesForRoundTrip(this.projectFileName), constants.yesString, constants.noString);
-			if (result === constants.yesString) {
-				await fs.copyFile(this._projectFilePath, this._projectFilePath + '_backup');
-				await this.updateSystemDatabaseReferencesInProjFile();
-			}
-		}
-	}
-
-	private async updateImportToSupportRoundTrip(): Promise<void> {
-		// update an SSDT project to include Net core target information
-		for (let i = 0; i < this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.Import).length; i++) {
-			const importTarget = this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.Import)[i];
-
-			let condition = importTarget.getAttribute(constants.Condition);
-			let projectAttributeVal = importTarget.getAttribute(constants.Project);
+		// TODO: is this check below still relevant?
+		// if (this._importedTargets.includes(constants.NetCoreTargets) && !this.containsSSDTOnlySystemDatabaseReferences() // old style project check
+		// 	|| this.sqlProjStyle === ProjectType.SdkStyle) { // new style project check
+		// 	return;
+		// }
 
-			if (condition === constants.SqlDbPresentCondition && projectAttributeVal === constants.SqlDbTargets) {
-				await this.updateImportedTargetsToProjFile(constants.RoundTripSqlDbPresentCondition, projectAttributeVal, importTarget);
-			}
-			if (condition === constants.SqlDbNotPresentCondition && projectAttributeVal === constants.MsBuildtargets) {
-				await this.updateImportedTargetsToProjFile(constants.RoundTripSqlDbNotPresentCondition, projectAttributeVal, importTarget);
-			}
-		}
-
-		await this.updateImportedTargetsToProjFile(constants.NetCoreCondition, constants.NetCoreTargets, undefined);
-	}
-
-	private async updateBeforeBuildTargetInProjFile(): Promise<void> {
-		// Search if clean target already present, update it
-		for (let i = 0; i < this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.Target).length; i++) {
-			const beforeBuildNode = this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.Target)[i];
-			const name = beforeBuildNode.getAttribute(constants.Name);
-			if (name === constants.BeforeBuildTarget) {
-				return await this.createCleanFileNode(beforeBuildNode);
-			}
-		}
+		TelemetryReporter.sendActionEvent(TelemetryViews.ProjectController, TelemetryActions.updateProjectForRoundtrip);
 
-		// If clean target not found, create new
-		const beforeBuildNode = this.projFileXmlDoc!.createElement(constants.Target);
-		beforeBuildNode.setAttribute(constants.Name, constants.BeforeBuildTarget);
-		this.projFileXmlDoc!.documentElement.appendChild(beforeBuildNode);
-		await this.createCleanFileNode(beforeBuildNode);
-	}
+		const result = await this.sqlProjService.updateProjectForCrossPlatform(this.projectFilePath);
```

Comment:
There should still be a prompt to ask the user if they want to update their project before actually updating it

Reply:
Kinda like throwing an error vs. calling `showErrorMessage()`, I figured `projectsController.ts` would do the confirmation before calling this method.  `projects.ts` shouldn't know/care about UI/inputs/prompts.

Reply:
...ooh, this is less clean than I was expecting due to dataWorkspace...